cmd/contour/serve.go

```diff
@@ -179,11 +196,19 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 
 	// step 4. register our resource event handler with the k8s informers.
 	var informers []cache.SharedIndexInformer
+	dynamicFactory := dynamicinformer.NewFilteredDynamicSharedInformerFactory(dynamicClient, 0, v1.NamespaceAll, nil)
+
+	dynamicClient.Resource(schema.GroupVersionResource{Group: "projectcontour.io", Version: "v1", Resource: "httpproxies"})
+	ingressRouteInformer := dynamicFactory.ForResource(schema.GroupVersionResource{Group: "contour.heptio.com", Version: "v1beta1", Resource: "ingressroutes"})
```

Comment:
Nit: the `GroupVersionResource` is used in two places, so it might be worth hoisting into constants in each API package. Your call, if you think it's worth it.